pkg/algorithm/cstorpoolselect/v1alpha1/select.go

```diff
@@ -133,51 +342,64 @@ func (p preferAntiAffinityLabel) filter(poolUIDs []string) ([]string, error) {
 type selection struct {
 	// list of original pools aginst whom
 	// selection will be made
-	poolUIDs []string
+	pools *csp.CSPList
 
 	// selection is based on these policies
-	policies []policy
+	policies *policyList
+
+	// mode flags if selection can consider
+	// multiple policies to select the pools
+	mode executionMode
 }
 
 // buildOption is a typed function that
 // abstracts configuring a selection instance
 type buildOption func(*selection)
 
+func withDefaultSelection(s *selection) {
+	if string(s.mode) == "" {
+		s.mode = singleExection
+	}
+}
+
 // newSelection returns a new instance of
 // selection
-func newSelection(poolUIDs []string, opts ...buildOption) *selection {
-	s := &selection{poolUIDs: poolUIDs}
+func newSelection(pools *csp.CSPList, mode executionMode, opts ...buildOption) *selection {
```

Comment:
Please remove `mode` as an argument.
The default is set via `withDefaultSelection`
It should be part of `buildOption` if caller code wants a specific mode.

Reply:
removed